### PR TITLE
Added task level log files and job tasks html report file

### PR DESCRIPTION
**Pull request by sunilkumarn417**
- Added task level log files.
- job tasks html report file in archive directory.

Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

### CODE REVIEW

**Review by rakeshgm — APPROVED** [review]

File: teuthology/run_tasks.py [hunk 1]
```diff
@@ -63,6 +68,88 @@ def _import(from_package, module_name, task_name, fail_on_import_error=False):
...
+
+def remove_log_handler(handler):
+    """
+    Remove log handler
+    Args:
+        handler: log handler obj
+    """
+    if handler:
+        root_logger.removeHandler(handler)
+
```
Comment (ryg-) [inline teuthology/run_tasks.py:108]: Probably should be added error processing section, minimally logging.

File: teuthology/run_tasks.py [hunk 2]
```diff
@@ -63,6 +68,88 @@ def _import(from_package, module_name, task_name, fail_on_import_error=False):
...
+def generate_unique_name(task_name, task_list):
+    """
+    Generate unique log file name
+    """
+    task_count = len([task_name for task in task_list
+                      if task.get('name').startswith(task_name)]) + 1
+    return "{}-{}.log".format(task_name, task_count), task_count
+
+
+def generate_result(all_tasks):
```
Comment (ryg-) [inline teuthology/run_tasks.py:140]: Do we really need prefix 'all_' here?

File: teuthology/run_tasks.py [hunk 3]
```diff
@@ -63,6 +68,88 @@ def _import(from_package, module_name, task_name, fail_on_import_error=False):
...
+    """
+    Display and Generate task results
+    """
+    mark = "=" * 25
```
Comment (ryg-) [inline teuthology/run_tasks.py:144]: I suggest to see 'results.py' e.g. line 233 and align marker format/length.

File: teuthology/run_tasks.py [hunk 4]
```diff
@@ -71,27 +158,70 @@ def run_one_task(taskname, **kwargs):
...
         timer = Timer()
     stack = []
+    all_tasks = []
     try:
         for taskdict in tasks:
             try:
                 ((taskname, config),) = taskdict.items()
             except (ValueError, AttributeError):
                 raise RuntimeError('Invalid task definition: %s' % taskdict)
+
```
Comment (ryg-) [inline teuthology/run_tasks.py:183]: I have feeling that there is too much support code added, which hides base idea of cycle. Probably, function should be created and log support code should be moved in it.

**Comment by ryg-** [comment]
@susebot run deploy

**Comment by rakeshgm** [comment]
@ryg- this is for `rh` branch, we will plan to submit when the full testing is completed.

**Comment by sunilkumarn417** [comment]
http://pulpito.ceph.redhat.com/sunnagar-2020-09-28_08:07:08-smoke:ceph-volume-lvm-master-distro-basic-clara/. 

failed test are due to connection issue, please ignore. but however both passed and failed results are captured.

**Tests**
Failed - http://magna002.ceph.redhat.com/sunnagar-2020-09-28_08:07:08-smoke:ceph-volume-lvm-master-distro-basic-clara/388925/
Passed - http://magna002.ceph.redhat.com/sunnagar-2020-09-28_08:07:08-smoke:ceph-volume-lvm-master-distro-basic-clara/388926/

**Pull request merged**